packages/fastify/src/ByuJwtProvider.ts

```diff
@@ -32,7 +32,7 @@ const ByuJwtProviderPlugin: FastifyPluginAsync<ByuJwtProviderOptions> = async (f
    */
   fastify.addHook('onRoute', (route) => {
     const foundPrefix = options.prefix != null
-    const matchesRoute = route.prefix === options.prefix
+    const matchesRoute = options.prefix != null && route.path.startsWith(options.prefix)
```

Comment:
```suggestion
    const matchesRoute = foundPrefix && route.path.startsWith(options.prefix)
```

Reply:
unfortunately typescript (or my IDE) doesn't auto-assume/infer that `options.prefix` is a string at this point. It says `Argument of type 'string | undefined' is not assignable to parameter of type 'string'`

Reply:
I'm in favor of removing the named variables then. Let's just do
```typescript
if (options.prefix != null && route.path.startsWith(options.prefix) {
  /** Don't add authentication to routes that don't start with the specified prefix */
  return 
}
```